Jason/pf project

JasonGibson274 opened the pull request:
This branch contains the particle filter project code.

The things left to do on this project is to 

1. Add additional noise to the odom measurement 
2. Correctly compute the transform of map to odom with additional noise

Otherwise this should be ready for general code review and reshuffling for the exercise.

I can assist when it comes to structuring the exercise around tuning of the particle filter.

barulicm committed:
Adds localization package.

barulicm committed:
Adds launch files to localization and frames out TF broadcasting.

barulicm committed:
Adds default value to launch configuration in PF launcher.

barulicm committed:
Merge remote-tracking branch 'origin/master' into dev/barulicm/pf_project

JasonGibson274 committed:
Merge branch 'master' into jason/pf_project

JasonGibson274 committed:
wrote out code for particle filter

JasonGibson274 committed:
Merge branch 'jason/coordinate_transform' into jason/pf_project

JasonGibson274 committed:
working a particle filter implmentation

JasonGibson274 committed:
fixed parameters not being set in particle filter

JasonGibson274 committed:
added separate timer for resampling particle filter

JasonGibson274 committed:
fixed motion model transformation in pf

JasonGibson274 committed:
reverted part of motion model change

JasonGibson274 committed:
fixed issue in aruco sensor model for incorrect yaw error

JasonGibson274 committed:
added error on vy in odometry model

JasonGibson274 committed:
working version of the IMU particle filter

JasonGibson274 committed:
moved callback internal to sensor models in pf

JasonGibson274 committed:
Merge branch 'main' into jason/pf_project

JasonGibson274 committed:
converted particle filter to use cmd_vel instead of IMU

JasonGibson274 committed:
Merge branch 'main' into jason/pf_project

JasonGibson274 committed:
cleaned up code and indicated student code in pf

barulicm committed:
Fixes linting failures.

barulicm committed:
Resolve TODO in particle_filter_localizer.cpp

barulicm committed:
Adds launch file for project 3

barulicm committed:
Refactors random number generation in localization.

barulicm committed:
Adds particle cloud to rviz config

barulicm committed:
Adds pose estimate to rviz config.

barulicm committed:
Adds aruco tag detector to week 3 launch.

JasonGibson274 committed:
fixed issue with invalid messages

coordinate frame is still flipped

JasonGibson274 committed:
Merge branch 'jason/pf_project' of github.com:RoboJackets/software-training into jason/pf_project

barulicm committed:
Adds coordinate transform node to week 3 launch.

barulicm committed:
Adds TF publication to localization node.

barulicm committed:
Various cleanups and fixes.

barulicm committed:
Add extra scripts to week 3 launch file and adjust rviz config.

barulicm committed:
More localization refactoring.

barulicm committed:
More minor cleanups.

barulicm committed:
Starter-code-ifies localization and does a bit more random cleanups.

barulicm committed:
Adds reference solution for localization.

barulicm committed:
Adds project 3 instructions.

barulicm committed:
Merge remote-tracking branch 'origin/main' into jason/pf_project

barulicm committed:
Fixes week 3 launch file.

barulicm commented:
Ok, @JasonGibson274, I've pushed up a bunch of changes including general cleanup, starter code changes, and project instructions.

Critical things I need your input on are in the instructions:
- Do we still want to do the tuning section? If so, I'll need your help with section 3.13.
- Do you have a good, one sentence explanation of what the normalizer is for the description of `ComputeLogNormalizer` in section 3.2?
- Double check my explanations of the theory related stuff in sections 3.8 and 3.9.

The changes here are dependent on the changes in [PR 25](https://github.com/RoboJackets/stsl/pull/25) in stsl. That PR will need to be merged and deployed for this project as well.

JasonGibson274 committed:
updated project 3 instructions for theory

added derivation of sensor update equations from univaraite and multivariate gaussian

JasonGibson274 committed:
fixed typos in week 3 instructions
